samples/core/xgboost_training_cm/xgboost_training_cm.py

```diff
@@ -201,86 +226,94 @@ def dataproc_predict_op(
     description='A trainer that does end-to-end distributed training for XGBoost models.'
 )
 def xgb_train_pipeline(
-    output,
-    project,
+    output='gs://guideline_example_bucket',
+    project='ml-pipeline-dogfood',
+    cluster_name='xgb-{{workflow.name}}',
     region='us-central1',
     train_data='gs://ml-pipeline-playground/sfpd/train.csv',
     eval_data='gs://ml-pipeline-playground/sfpd/eval.csv',
     schema='gs://ml-pipeline-playground/sfpd/schema.json',
     target='resolution',
-    rounds=200,
+    rounds=5,
     workers=2,
     true_label='ACTION',
 ):
     output_template = str(output) + '/' + dsl.EXECUTION_ID_PLACEHOLDER + '/data'
 
-    delete_cluster_op = dataproc_delete_cluster_op(
-        project,
-        region
-    ).apply(gcp.use_gcp_secret('user-gcp-sa'))
+    # Current GCP pyspark/spark op do not provide outputs as return values, instead,
+    # we need to use strings to pass the uri around.
+    analyze_output = output_template
+    transform_output_train = os.path.join(output_template, 'train', 'part-*')
+    transform_output_eval = os.path.join(output_template, 'eval', 'part-*')
+    train_output = os.path.join(output_template, 'train_output')
+    predict_output = os.path.join(output_template, 'predict_output')
+
+    with dsl.ExitHandler(exit_op=dataproc_delete_cluster_op(
+        project_id=project,
+        region=region,
+        name=cluster_name
+    ).apply(gcp.use_gcp_secret('user-gcp-sa'))):
+
+        # create_cluster_op = dataproc_create_cluster_op(
+        #     project=project,
+        #     region=region,
+        #     staging=output
+        # ).apply(gcp.use_gcp_secret('user-gcp-sa'))
 
-    with dsl.ExitHandler(exit_op=delete_cluster_op):
         create_cluster_op = dataproc_create_cluster_op(
-            project,
-            region,
-            output
+            project_id=project,
+            region=region,
+            name=cluster_name,
+            initialization_actions=[
+              os.path.join(_PYSRC_PREFIX,
+                           'create/initialization_actions.sh'),
+            ],
+            image_version='1.2'
         ).apply(gcp.use_gcp_secret('user-gcp-sa'))
 
         analyze_op = dataproc_analyze_op(
-            project,
-            region,
-            create_cluster_op.output,
-            schema,
-            train_data,
-            output_template
-        ).apply(gcp.use_gcp_secret('user-gcp-sa'))
+            project=project,
+            region=region,
+            cluster_name=cluster_name,
+            schema=schema,
+            train_data=train_data,
+            output=output_template
+        ).after(create_cluster_op).apply(gcp.use_gcp_secret('user-gcp-sa'))
```

Comment:
It might be nicer to apply the secrets to all ops in one line:
```python
kfp.dsl.get_pipeline_conf().add_op_transformer(use_gcp_secret('user-gcp-sa'))
```

Reply:
Good point